pkg/licensing/license_manager.go

```diff
@@ -0,0 +1,100 @@
+package licensing
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/lib/license"
+)
+
+// LicenseManager handles license management and validation for the node
+type LicenseManager struct {
+	bacalhauConfig    *types.Bacalhau
+	licenseConfigured bool
+	licenseToken      string
+	licenseValidator  *license.LicenseValidator
+}
+
+// NewLicenseManager creates and initializes a new LicenseManager
+func NewLicenseManager(config *types.Bacalhau) (*LicenseManager, error) {
+	if config == nil {
+		return nil, errors.New("config cannot be nil")
+	}
+
+	validator, err := license.NewOfflineLicenseValidator()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create license validator: %w", err)
+	}
+
+	licensePath := config.Orchestrator.License.LocalPath
+	licenseConfigured := licensePath != ""
+
+	var licenseToken string
+	if licenseConfigured {
+		// Try to read the license file, fail if the file is not found or malformed
+		// Will not inspect the license itself.
+		licenseData, err := os.ReadFile(licensePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read license file: %w", err)
+		}
+
+		// Verify the JSON structure
+		var licenseFile struct {
+			License string `json:"license"`
+		}
```

Comment:
I don't like having inline structs for such behaviour. First, why can't we have license content at the root of the json object instead of having a `license` field? If it is a must, then lets introduce a `LicenseWrapper` type and not make it inline

Reply:
@wdbaruni 

> First, why can't we have license content at the root of the json object instead of having a license field?
Not sure what you meant here, but the license content is already at the root of the json object. The Json file structure of the license:

```json
{
  "license": "the-license-jwt-token"
}
```

> If it is a must, then lets introduce a LicenseWrapper type and not make it inline
Hmm, It is not used anywhere else, do you think it is still worth it ? 
The unit tests already cover all the edge cases for the format.

Reply:
if the file only contains the license jwt token and nothing else, why do you need a json wrapping the token?

Reply:
@wdbaruni I would rather have a JSON object as the structure of a license even if it has one item in it. It will easily allow for any future additions or migrations, compared to just an opaque string.